extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -178,6 +179,23 @@ private void registerModuleIfOnClassPath(String moduleClassName,
     @Record(ExecutionTime.STATIC_INIT)
     SyntheticBeanBuildItem pushConfigurationBean(JacksonRecorder jacksonRecorder,
             JacksonBuildTimeConfig jacksonBuildTimeConfig) {
+
+        if (jacksonBuildTimeConfig.timezone.isPresent()) {
+            /*
+             * We need to make timezone a String instead of a java.util.TimeZone class
+             * because:
+             * 1) TimeZone cannot automatically be handled by the BytecodeRecorder
+             * 2) Handling it would require us to add non-JDK classes (i.e. sun.util.calendar.ZoneInfo) to bytecode recording
+             */
+            String timeZoneStr = jacksonBuildTimeConfig.timezone.get();
+            TimeZone timeZone = TimeZone.getTimeZone(timeZoneStr);
+            if ("GMT".equals(timeZone.getID()) && !timeZoneStr.startsWith("GMT")) {
```

Comment:
Open question: I wonder if we should use `ZoneId` instead and only convert it to `TimeZone` when pushing it to Jackson. It seems to have a saner behavior when parsing (and there is a `checkAvailable` boolean that might make sense to use).

Reply:
That's interesting. Let me explore

Reply:
I pushed a second commit that introduces support for `ZoneId`